docs/data/guides/telemetry/telemetry.md

```diff
@@ -0,0 +1,91 @@
+---
+packageName: '@mui/x-telemetry'
+---
+
+# MUI X Telemetry guide
+
+<p class="description">MUI X Telemetry collects anonymous usage data to help improve the library. This guide will walk you through how to opt-in, opt-out, and configure telemetry.</p>
+
+## Opting In
```

Comment:
@joserodolfofreitas It would be great to have a few paragraphs here explaining what's the purpose of telemetry and why would users want to enable it.

Reply:
I'll address this separately together with the blog post to announce telemetry to the public. For now, we can merge this for the internal testing.